php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -158,22 +158,22 @@ class NDB_BVL_Instrument extends NDB_Page
      * $instrument, and runs the setup() method on that new
      * instrument.
      *
-     * @param string  $instrument       The name of the instrument to use
-     * @param string  $commentID        The CommentID identifying the data to load
-     *                                  load
-     * @param string  $page             If a multipage form, the page to show
-     * @param boolean $guarantee_exists If true the factory will throw an
-     *                                  error if the file does not exist, otherwise
-     *                                  will silently return if the instrument is
-     *                                  missing.
+     * @param string      $instrument       The name of the instrument to use
+     * @param string      $commentID        The CommentID identifying the data to
+     *                                      load
+     * @param string|null $page             If a multipage form, the page to show
+     * @param boolean     $guarantee_exists If true the factory will throw an
+     *                                      error if the file does not exist,
+     *                                      otherwise will silently return if the
+     *                                      instrument is missing.
      *
      * @return object|null the new object of $instrument type
      * @access public
      */
     static function factory(
         string $instrument,
         string $commentID = '',
-        string $page = '',
+        ?string $page = null,
```

Comment:
I don't think it makes sense to make this nullable, it just leads to confusion between the empty string and null. Why not just use the null coalesce operator with '' at the call site to fix the caller?

Reply:
Hmm yeah that's probably a better solution.